# Total planning attempts
max_solutions: 10

# Planning group and link names
arm_group_name: "panda_arm"
#eef_name: "hand"
eef_name: "panda_hand"
#hand_group_name: "hand"
hand_group_name: "panda_hand"
# hand_frame: "panda_link8"
hand_frame: "panda_hand"

# Poses
hand_open_pose: "open"
hand_close_pose: "close"
arm_home_pose: "ready"

# Scene frames
world_frame: "world"
table_reference_frame: "world"
# object_reference_frame: "world"
object_reference_frame: "panda_link0"
surface_link: "table"

# Collision object for picking
# CYLINDER object specifications
object_name: "object"
object_dimensions: [0.064, 0.01] # [height, radius]
object_pose: [0.5, 0, 0.443171, 0, 0, 0]

# Collision object for assembly
# CYLINDER object specifications
assembly_object_name: "assembly_object"
assembly_object_dimensions: [0.064, 0.01] # [height, radius]
assembly_object_pose: [0.5, 0.2, 0.443171, 0, 0, 0]

# Table model
spawn_table: true
table_name: "table"
table_dimensions: [0.453157, 0.642049, 0.443171]  # [length, width, height]
table_pose: [0.5, 0.0, 0.443171, 0, 0, 0]

# Gripper grasp frame transform [x,y,z,r,p,y]
# grasp_frame_transform: [0, 0, 0.1, 1.571, 0.785, 1.571]
# grasp_frame_transform: [0, 0, 0.1, 0, 1.571, 0]
grasp_frame_transform: [0, 0, 0.1, 0, 3.14, 0]

# Place pose [x,y,z,r,p,y]
# place_pose: [0.6, -0.15, 0, 0, 0, 0]
place_pose: [0.5, 0.3, 0.52, 0, 0, 0]
# place_pose: [0.5, 0.3, 0.532, 0, 0, 0]
place_surface_offset: 0.0001 # place offset from table

# Valid distance range when approaching an object for picking
approach_object_min_dist: 0.1
approach_object_max_dist: 0.15

# Valid height range when lifting an object after pick
lift_object_min_dist: 0.01
lift_object_max_dist: 0.1
